lib/mixlib/install/backend/bintray.rb

```diff
@@ -0,0 +1,210 @@
+        #
+        def bintray_artifacts
+          version = options.product_version == :latest ? latest_version : options.product_version
+          results = bintray_get("versions/#{version}/files")
+
+          # Delete .asc files
+          results.each { |r| results.delete(r) if r["name"].end_with?(".asc") }
+
+          # Convert results to build records
+          results.map { |a| create_artifact(a) }
+        end
+
+        #
+        # Creates an instance of ArtifactInfo
+        #
+        # @param artifact_map
+        # {
+        #   "name" => "chef-12.8.1-1.powerpc.bff",
+        #   "path" => "aix/6.1/chef-12.8.1-1.powerpc.bff",
+        #   "version" => "12.8.1",
+        #   "sha1" => "1206f7be7be8bbece1e9943dcdc0d22fe538718b",
+        #   "sha256" => "e49321095a04f51385a59b3f3d7223cd1bddefc2e2f4280edfb0934d00a4fa3f"
+        # }
+        #
+        # @return [ArtifactInfo] ArtifactInfo instance
+        #
+        def create_artifact(artifact_map)
+          platform_info = parse_platform_info(artifact_map)
+
+          url = "#{DOWNLOAD_URL_ENDPOINT}/#{options.channel}/#{artifact_map["path"]}"
+
+          ArtifactInfo.new(
+            sha1:             artifact_map["sha1"],
+            sha256:           artifact_map["sha256"],
+            version:          artifact_map["version"],
+            platform:         platform_info[:platform],
+            platform_version: platform_info[:platform_version],
+            architecture:     platform_info[:architecture],
+            url:              url
+          )
+        end
+
+        #
+        # Parses platform info
+        #
+        # @param artifact_map
+        # {
+        #   "name" => "chef-12.8.1-1.powerpc.bff",
+        #   "path" => "aix/6.1/chef-12.8.1-1.powerpc.bff",
+        #   "version" => "12.8.1",
+        #   "sha1" => "1206f7be7be8bbece1e9943dcdc0d22fe538718b",
+        #   "sha256" => "e49321095a04f51385a59b3f3d7223cd1bddefc2e2f4280edfb0934d00a4fa3f"
+        # }
+        #
+        # @return [Hash] platform, platform_version, architecture
+        #
+        def parse_platform_info(artifact_map)
+          # platform/platform_version/filename
+          path = artifact_map["path"].split("/")
+          platform = path[0]
+          platform_version = path[1]
+
+          filename = artifact_map["name"]
+          architecture = if %w{ x86_64 amd64 x64 }.fuzzy_include?(filename)
+                           "x86_64"
+                         elsif %w{ i386 x86 }.fuzzy_include?(filename)
+                           "i386"
+                         elsif %w{ powerpc }.fuzzy_include?(filename)
+                           "powerpc"
+                         elsif %w{ sparc }.fuzzy_include?(filename)
+                           "sparc"
+                         elsif platform == "mac_os_x"
+                           "x86_64"
+                         else
+                           "unknown"
```

Comment:
Would it be better to raise here with an `UnknownArchitecture` exception or similar?